src/applications/vaos/redux/reducer.js

```diff
@@ -1,7 +1,8 @@
 import appointments from '../appointment-list/redux/reducer';
+import newAppointmentReducer from '../new-appointment/redux/reducer';
 
 export default {
-  // newAppointment state slice lazy loaded (new-appointment/redux/reducer.js)
   // expressCare state slice lazy loaded (express-care/redux/reducer.js)
   appointments,
+  newAppointment: newAppointmentReducer,
```

Comment:
I'm curious how does adding `newAppointment: newAppointmentReducer` in this reducer file affect the out-of-flow redirect?

Reply:
Yeah this took me a few to figure out...Do you remember yesterday when I showed you the "Start schedule" button click event where we where dispatching the "start new appointment" action and then routing to the new-appointment component. What I found was....there was no newAppointment object in the store for the dispatch to change. That object gets initialized when we route to the new-appointment component. By putting this in the base reducer we now have an initialized newApointment for the dispatch to work. Here is the button click event
https://github.com/department-of-veterans-affairs/vets-website/blob/88875c94508f234c077dbca5c62916893772e447/src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx#L12-L20
 

And here is where the newApointment object gets added to the store in the route
https://github.com/department-of-veterans-affairs/vets-website/blob/88875c94508f234c077dbca5c62916893772e447/src/applications/vaos/routes.jsx#L36-L43

I think we never really had to rely on that sequence until now....

Reply:
Oh yes, I remember now. Great catch!